pandas/core/indexes/datetimes.py

```diff
@@ -2239,6 +2239,27 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
     Returns
     -------
     rng : DatetimeIndex
+
+    Examples
+    --------
+    >>> pd1 = pd.date_range(start='1/1/2018', end='2/1/2018')
+    >>> pd1
+    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
+               '2018-01-05', '2018-01-06', '2018-01-07', '2018-01-08',
+               '2018-01-09', '2018-01-10', '2018-01-11', '2018-01-12',
+               '2018-01-13', '2018-01-14', '2018-01-15', '2018-01-16',
+               '2018-01-17', '2018-01-18', '2018-01-19', '2018-01-20',
+               '2018-01-21', '2018-01-22', '2018-01-23', '2018-01-24',
+               '2018-01-25', '2018-01-26', '2018-01-27', '2018-01-28',
+               '2018-01-29', '2018-01-30', '2018-01-31', '2018-02-01'],
```

Comment:
can you add an example using ``freq`` and ``tz`` (separate or together is ok), and one with ``closed`` (and its variations)